### PR TITLE
feat(robot): add options for robot teammembership + typed team responses

### LINKED ISSUE #54: `teams` and `relationships` is missing
### What problem are you facing?

Currently, it's not feasible to utilize up-sdk-go for configuring `teams` and `relationships`.
Due to this limitation, we must create our own client and methods within provider-upbound.

https://github.com/upbound/provider-upbound/blob/main/internal/client/teams/client.go
https://github.com/upbound/provider-upbound/blob/main/internal/client/robotteammembership/client.go

### How could Upbound help solve your problem?

Epic #

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
- add create / delete for robot team memberships
- typed team response instead of generic DataSet to ex. extract names

Fixes #54

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested
via up-cli PR:

```
up robot membership create chris test123 -a upboundcare
upboundcare/chris/test123 team membership created
```

image

```
up robot membership get chris -a upboundcare    
TEAM MEMBERSHIP
test444
test123
```

```
up robot membership delete chris test123 -a upboundcare
Are you sure you want to delete this robot team membership? This cannot be undone [y/n]: y
Deleting robot team membership upboundcare/chris/test123.
upboundcare/chris/test123 deleted
```

```
up robot membership get chris -a upboundcare
TEAM MEMBERSHIP
test444
```

### CODE REVIEW

File: service/robots/client.go [hunk 1]
```diff
@@ -96,3 +97,25 @@ func (c *Client) Delete(ctx context.Context, id uuid.UUID) error { // nolint:int
 	}
 	return c.Client.Do(req, nil)
 }
+
+// CreateTeamMembership create a robot team membership on Upbound.
+func (c *Client) CreateTeamMembership(ctx context.Context, id uuid.UUID, params *RobotTeamMembershipResourceIdentifier) error {
```
Comment (tnthornton) [inline service/robots/client.go:102]: Does params need to be a pointer? I'm curious a double pointer when adding it to the RelationshipList below 🤔

Comment (haarchri) [reply service/robots/client.go:102]: What is different to the other create functions ? List is only possible via robot get

Comment (haarchri) [reply service/robots/client.go:102]: Does it help to open the up-cli PR to show the implementation?

Comment (haarchri) [reply service/robots/client.go:102]: https://github.com/upbound/up/pull/568

Comment (tnthornton) [reply service/robots/client.go:102]: Still feel like this is a little weird needing to jump through the pointer hoops when we control the contract, but I won't block on it.

**Review by tnthornton — APPROVED** [review]

**Pull request merged**